### PR TITLE
Docs: Update the Style Guide

### LINKED ISSUE #1249: Style Guide Update
**What should be documented? Is there something wrong in the documentation?**
The Style guide needs to be update based on new standards that came about as per the client manaual

**Do you have suggestions?**

style guide - usage of contractions such as can't and don't should be avoided, also think about incorproating some of https://devguide.python.org/documentation/style-guide/ into it, especially capitalization rules. another source to pull from - https://docs.gitlab.com/ee/development/documentation/styleguide/

style guide - usage of the oxford comman

style guide - capitalization rules
* Title Headings
* Menu Entries
* Anything inside `:guilabel:` or `:menuselection:` tags
* Highlighted Game Elements:

> * Units -- `:unit:` tag
> * Improvements -- `:improvement:` tag
> * Wonders -- `:wonder:` tag

style guide - usage of `numfig`

### PR DESCRIPTION
**Pull request by jwrober**
Closes #1249

### CODE REVIEW

**Review by lmoureaux — COMMENTED** [review]
Some more items that we could include:

* Which person to use ("you can push the button", "the button can be pushed", "one can push the button"...)
* Discouraging double negations ("the guidelines aren't bad" -> "the guidelines are good")

File: docs/Contributing/style-guide.rst [hunk 1]
```diff
@@ -115,3 +127,54 @@ matching the admonition type. We use some of the standard admonitions in our doc
...
+    at the end of the list. For example: You need to follow these steps: Click on :guilabel:`Menu`, then click
+    on :guilabel:`Options`, and finally click on :guilabel:`Interface`. Notice the comma usage before the word
+    "and", that is the Oxford command and its usage is expeced in our documentation
+
+Capitalization
+    For consistent formatting, the following should always use "Title Capitalization" rules:
+
+    * Section Headings (e.g. the 4 documented above)
+    * Menu Entries in the client
+    * Anything inside of :literal:`:guilabel:` or :literal:`:menuselection:`
```
[the remark this answers is not in the document]
Comment (jwrober) [reply docs/Contributing/style-guide.rst:152]: As we discussed in Discord, I thinks both ways. We should use title case in gui elements and menus.

File: docs/Contributing/style-guide.rst [hunk 2]
```diff
@@ -115,3 +127,54 @@ matching the admonition type. We use some of the standard admonitions in our doc
...
+    * The names of Units, City Improvements and Wonders in the :literal:`:unit:`, :literal:`:improvement:` and
+      :literal:`:wonder:` tags respectively
+
+Language Contractions
+    Language Contractions are when two words are combined together with an apostrophe ( ``'`` ). For example,
+    the word "don't" is a contraction of "do not". Not all language translators, and especially non-native
```
[the remark this answers is not in the document]
Comment (jwrober) [reply docs/Contributing/style-guide.rst:161]: Yes that is true, but I have also been told that they suck for the Asian languages and I think some of the Cyrillic languages as well.

File: docs/Contributing/style-guide.rst [hunk 3]
```diff
@@ -115,3 +127,54 @@ matching the admonition type. We use some of the standard admonitions in our doc
...
+    English speakers can get confused if contractions are used. To aid the readability of our documentation,
+    :strong:`the usage of contractions is not acceptable`.
```
[the remark this answers is not in the document]
Comment (jwrober) [reply docs/Contributing/style-guide.rst:163]: Right exactly, so I think keeping contractions at a minimum is a good practice.

**Pull request merged**